src/vs/workbench/services/extensionManagement/common/extensionManagementService.ts

```diff
@@ -269,7 +279,9 @@ export class ExtensionManagementService extends Disposable implements IWorkbench
 					servers.push(this.extensionManagementServerService.localExtensionManagementServer);
 				}
 			}
-			return Promises.settled(servers.map(server => server.extensionManagementService.installFromGallery(gallery, installOptions))).then(([local]) => local);
+			return this.promptForTrustIfNeededAndInstall(manifest, () => {
+				return Promises.settled(servers.map(server => server.extensionManagementService.installFromGallery(gallery, installOptions))).then(([local]) => local);
+			});
```

Comment:
I would decouple `promptForTrustIfNeeded` and `install` operations for simplification, say:

```
await this.checkForTrustedWorkspace();
return Promises.settled(servers.map(server => server.extensionManagementService.installFromGallery(gallery, installOptions))).then(([local]) => local);
```

Reply:
Fixed. Can you please have another look? Thanks!